Subject: ieee1394-skip-unnecessary-pause-when-scanning-config-roms
From: Jody McIntyre <scjody@steamballoon.com>

ieee1394/nodemgr:

Skip a superfluous pause that occured when the config ROM of a node was
scanned unsuccessfully.  This also occurs if a node without link wrongly
enables its "link active" self ID flag.  A GWCTech 6-port hub does this.

Signed-off-by: Stefan Richter <stefanr@s5r6.in-berlin.de>
Signed-off-by: Jody McIntyre <scjody@steamballoon.com>
Cc: Ben Collins <bcollins@debian.org>
Signed-off-by: Andrew Morton <akpm@osdl.org>
Signed-off-by: Olaf Hering <olh@suse.de>
---

 drivers/ieee1394/nodemgr.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff -puN drivers/ieee1394/nodemgr.c~ieee1394-skip-unnecessary-pause-when-scanning-config-roms drivers/ieee1394/nodemgr.c
--- 25/drivers/ieee1394/nodemgr.c~ieee1394-skip-unnecessary-pause-when-scanning-config-roms	Thu Sep 22 15:07:14 2005
+++ 25-akpm/drivers/ieee1394/nodemgr.c	Thu Sep 22 15:07:14 2005
@@ -64,10 +64,10 @@ static int nodemgr_bus_read(struct csr12
 	struct nodemgr_csr_info *ci = (struct nodemgr_csr_info*)__ci;
 	int i, ret = 0;
 
-	for (i = 0; i < 3; i++) {
+	for (i = 1; ; i++) {
 		ret = hpsb_read(ci->host, ci->nodeid, ci->generation, addr,
 				buffer, length);
-		if (!ret)
+		if (!ret || i == 3)
 			break;
 
 		if (msleep_interruptible(334))
_
